### ISSUE TITLE
Mouse Events not caputred in InfoWindow Widgets

### ISSUE THREAD

**Issue by umpire003** [branflake2267/GWT-Maps-V3-Api#183]
In versions 3.9 and 3.10, I have encountered an issue where button click and mouse events are not captured on Widgets added to InfoWindows.  The InfoWindow displays properly, the button is added correctly, however events are not captured.  Below is a snippet from the my code.  Please advise if there is better way to capture these events.

InfoWindowOptions options = InfoWindowOptions.newInstance();
if (infoWindow != null) {
    infoWindow.close();
}

infoWindow = InfoWindow.newInstance(options);

Button btn = new Button("<img src='images/printer.png' border='0'/>");
btn.addClickHandler(new ClickHandler() {
    @Override
    public void onClick(ClickEvent event) {
        String test = "test";
        com.google.gwt.user.client.Window.alert("click");
    }
});
btn.addMouseDownHandler(new MouseDownHandler() {
    @Override
    public void onMouseDown(MouseDownEvent event) {
        com.google.gwt.user.client.Window.alert("mouse down");
    }
});

infoWindow.setContent(btn);
infoWindow.open(map, marker);

**Comment by branflake2267** [comment]
huh?

**Comment by umpire003** [comment]
Sorry, I have updared the issue... saved it by accident. My apologies!

**Comment by branflake2267** [comment]
np, was wondering :), did you take a look at the examples yet, I can't remember what I did at the moment. :)

**Comment by umpire003** [comment]
Yes, I looked at the examples, and believe that I've followed them, however I'm definitely missing something.  I'll keep looking.  I am trying to rebuild a Google Maps V2 app using your V3 libraries.

Running example with out working button clicks:
http://apps.esportscenter.net/boundariesnew/?host=test&D=8

Example of working V2 functionality:
http://www.findmyleague.com/western-region/little-league-northwest-region/washington-boundary-maps/western-washington/washington-district-8.html

**Comment by branflake2267** [comment]
yeah, I can't remember if I added the widget feature to tooltip, its possible I forgot about that. I did add it to the controls on the map. Hm.

**Comment by branflake2267** [comment]
Nice map by the way.

**Comment by branflake2267** [comment]
Look for MapPanel or something of that name, add your widget to that and add the map panel to the info widget. I'm not sure that will work yet, I'm not able to look yet. :)

**Comment by umpire003** [comment]
I was able to get button click events to be recognized correctly by calling the InfoWindowOptions.setContent(Widget) method instead of calling the InfoWindow.setContent(Widget) method.

**Comment by treethinker** [comment]
I experienced this, too, by the way. There are a couple of places where both
XobjectOptions.setBehavior() and Xobject.setBehavior() exist, and the Xobject.setBehavior()
does not work. It would seem me to clarify the API a lot to just confine it so that you had
to use the Options object for all behaviors.

Another example is MapWidget.setMapTypeId() versus MapOptions.setMapTypeId() 

I am grateful for your work, so I am hardly screaming for this, because I've sorted through it,
but I believe it would speed up development to make this clear.

**Comment by branflake2267** [comment]
Good to know. Thanks for sharing. :)
